### ISSUE TITLE
wesnoth.interface.select_unit(nil) unexpected behavior

### ISSUE THREAD

**Issue by hladkyjakub** [wesnoth/wesnoth#9213]
### Game and System Information

- **Version:** 1.18.0

### Description of the bug

1. Visual bug when deselecting unit `wesnoth.interface.select_unit(nil,nil,false)`, **while** selected unit is being hovered over, **once the mouse moves out** black stripes are still all over whole screen (at this point unit is unselected, and mouse is on other tile).
Doesn't happen when the mouse is elsewhere, viz video

On the video I first select the unit, and move the mouse away from unit, everything works fine after unselecting unit in Lua console.
Then I select the unit again, while keeping the **mouse cursor hovering over the unit**, then I unselect the unit via Lua console, but then there are those black stripes all over the screen **when I move the mouse cursor away**, which shouldn't be.
https://github.com/user-attachments/assets/ea40665b-7fe5-49c0-b873-df1a26aca41a

2. `wesnoth.interface.select_unit(nil,nil,false)` Unselecting unit in wesnoth.game_events.on_mouse_button does nothing.

### Steps to reproduce the behavior

1. 1. select a unit
1. 2. hover over an empty hex
1. 3. Into Lua console: `wesnoth.interface.select_unit(nil,nil,false`
1. 4.  ou should now see a map, like normally, when no unit is selected **even when you move mouse**
1. 5. select the unit
1. 6. hover over the exact same unit
1. 7. Into Lua console: `wesnoth.interface.select_unit(nil,nil,false`
1. 8. you should now see a map, like the unit that you unselected in Lua would be selected **even when you move a mouse** BUG (Including black stripes)

-2. 1. Into Lua console:
```
wesnoth.game_events.on_mouse_button = function(x, y, button, event)
    if not (button == "left" and event == "click") then
        return false
    end
    if not(wesnoth.interface.get_selected_hex()) then
        return false
    end
    local unit = wesnoth.units.get(wesnoth.interface.get_selected_hex() )
    if unit.type~="Mage" then
        return false
    end
    wesnoth.interface.select_unit(nil)
    wesnoth.interface.add_chat_message("Mage was selected")
    return true
end
```
-2. 2. In UI:
Select a mage.

### Expected behavior

1. no gray screen, upon deselecting unit, while it's being hovered over.

2. deselected unit.

**Comment by jy02140486** [comment]
I try it in 1.19, doesn't happen
The c++ impl for this command should be int game_lua_kernel::intf_select_unit(lua_State *L) I think
but selecting a unity by mouse click doesn't run into this function

**Comment by CelticMinstrel** [comment]
Yes, that function only runs when Lua code asks to select a unit.

**Comment by CelticMinstrel** [comment]
Fixed by #9446

**Comment by babaissarkar** [comment]
#9446 was rebased merged from command line so the status is showing up as "Closed" instead of "Merged".

**Comment by hladkyjakub** [comment]
> #9446 was rebased merged from command line so the status is showing up as "Closed" instead of "Merged".
> > The merged commits are 0c2b035 and 88c6c13fddd6430e27da3f2c5b672539a600e91d.

Nice it was merged...
Why was it merged from command line?
